src/render/OpenGL.cpp

```diff
@@ -1058,17 +1058,21 @@ void CHyprOpenGLImpl::renderBorder(wlr_box* box, const CGradientValueData& grad,
     static auto* const PBORDERSIZE  = &g_pConfigManager->getConfigValuePtr("general:border_size")->intValue;
     static auto* const PMULTISAMPLE = &g_pConfigManager->getConfigValuePtr("decoration:multisample_edges")->intValue;
 
-    wlr_box            newBox = *box;
+    if (borderSize < 0) {
+        borderSize = *PBORDERSIZE;
```

Comment:
I don't like this approach, I'd rather make borderSize required and NOT shadow it here with a config var, instead doing the checks and stuff in the renderer.

Reply:
fair enough. I hadn't checked where else this code was used so I just went for the variable with default, trying not to break anything else